tests/unit/s2n_connection_serialize_test.c

```diff
@@ -592,6 +592,70 @@ int main(int argc, char **argv)
         }
     };
 
+    /* Self-talk: A deserialized session can be serialized */
+    for (size_t i = 0; i < s2n_array_len(config_array); i++) {
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(client_conn);
+        DEFER_CLEANUP(struct s2n_connection *first_server = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(first_server);
+
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config_array[i]));
+        EXPECT_SUCCESS(s2n_connection_set_config(first_server, config_array[i]));
+
+        DEFER_CLEANUP(struct s2n_test_io_stuffer_pair io_pair = { 0 }, s2n_io_stuffer_pair_free);
+        EXPECT_OK(s2n_io_stuffer_pair_init(&io_pair));
+        EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, first_server, &io_pair));
+
+        EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(first_server, client_conn));
+
+        /* Preliminary send and receive */
+        EXPECT_OK(s2n_send_and_recv_test(first_server, client_conn));
+        EXPECT_OK(s2n_send_and_recv_test(client_conn, first_server));
+
+        /* First serialization */
+        uint8_t first_serialization[S2N_SERIALIZED_CONN_TLS12_SIZE] = { 0 };
+        EXPECT_SUCCESS(s2n_connection_serialize(first_server, first_serialization, sizeof(first_serialization)));
+
+        /* Second deserialization */
+        DEFER_CLEANUP(struct s2n_connection *second_server = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(second_server);
+        EXPECT_SUCCESS(s2n_connection_deserialize(second_server, first_serialization, sizeof(first_serialization)));
+
+        /* Wipe and re-initialize IO pipes */
+        EXPECT_SUCCESS(s2n_stuffer_wipe(&io_pair.client_in));
+        EXPECT_SUCCESS(s2n_stuffer_wipe(&io_pair.server_in));
+        EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, second_server, &io_pair));
+
+        /* Another send and recv */
+        EXPECT_OK(s2n_send_and_recv_test(second_server, client_conn));
+        EXPECT_OK(s2n_send_and_recv_test(client_conn, second_server));
+
+        /* Second serialization. Note that the server needs a config with the serialization version set. */
```

Comment:
Didn't the server need a config with the serialization version set the first time too? We're using the same config. I might not be understanding what this comment is trying to warn me about.